assets/js/phoenix.js

```diff
@@ -940,12 +940,46 @@ export class Socket {
   }
 
   teardown(callback, code, reason){
-    if(this.conn){
-      this.conn.onclose = function(){} // noop
-      if(code){ this.conn.close(code, reason || "") } else { this.conn.close() }
-      this.conn = null
+    if (!this.conn) {
```

Comment:
is this if needed? I see similar checks in waitForBufferDone and waitForSocketClosed

Reply:
Yep we don't want to kick off the process if no `conn`